internal/core/processor/universal/updater/updater.go

```diff
@@ -0,0 +1,248 @@
+package updater
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/freeverseio/laos-universal-node/internal/platform/model"
+	"github.com/freeverseio/laos-universal-node/internal/platform/scan"
+	"github.com/freeverseio/laos-universal-node/internal/platform/state"
+)
+
+const historyLength = 256
+
+type Updater interface {
+	GetModelTransferEvents(
+		ctx context.Context,
+		startingBlock,
+		lastBlock uint64,
+		contracts []string,
+	) (map[string][]model.ERC721Transfer, error)
+
+	UpdateStates(
```

Comment:
`UpdateState` seems more appropriate to me, we only have one state

Reply:
ok